mmedit/models/editors/guided_diffusion/adm.py

```diff
@@ -0,0 +1,326 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from copy import deepcopy
+from typing import Dict, List, Optional, Union
+
+import mmengine
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmengine import MessageHub
+from mmengine.model import BaseModel, is_model_wrapper
+from mmengine.optim import OptimWrapperDict
+from mmengine.runner.checkpoint import _load_checkpoint_with_prefix
+from tqdm import tqdm
+
+from mmedit.registry import DIFFUSERS, MODELS, MODULES
+from mmedit.structures import EditDataSample, PixelData
+from mmedit.utils.typing import ForwardInputs, SampleList
+
+ModelType = Union[Dict, nn.Module]
+
+
+@MODELS.register_module('ADM')
+@MODELS.register_module('GuidedDiffusion')
+@MODELS.register_module()
+class AblatedDiffusionModel(BaseModel):
+    """Ablated diffusion model mentioned in the paper Diffusion Models Beat
+    GANs on Image Synthesis (https://arxiv.org/pdf/2105.05233.pdf).
+
+    Args:
+        data_preprocessor (dict, optional): The pre-process config of
+            :class:`BaseDataPreprocessor`.
+        unet (Union[Dict, nn.Module]): The config or the model of denoising
+            unet.
+        diffuser (Dict): The config of the diffusion scheduler.
+        use_fp16 (bool, optional): Whether to use fp16 parameters for blocks
+            of unet. Defaults to False.
+        classifier (Optional[Union[Dict, nn.Module]]): The config or the model
+            of the classifier. Defaults to None.
+        pretrained_cfgs (Optional[Dict]): The configs of the pretrained
+            weights. Defaults to None.
+    """
+
+    def __init__(self,
+                 data_preprocessor,
+                 unet,
+                 diffuser,
+                 use_fp16=False,
+                 classifier=None,
+                 classifier_scale=1.0,
+                 pretrained_cfgs=None):
+
+        super().__init__(data_preprocessor=data_preprocessor)
+        self.unet = MODULES.build(unet)
+        self.diffuser = DIFFUSERS.build(diffuser)
+        if classifier:
+            self.classifier = MODULES.build(unet)
```

Comment:
The classifier uses the exactly same config with unet. Will this cause the classifier to have the same output as the unet model, which is 6 dimensions? But the classifier should have the output with dimension number which is the number of classes (e.g. 1000 for imagenet?)

Reply:
Thanks for your comments. This is a typo mistake. Fix soon.